### PR TITLE
Add nghttp2 recipe

### PR DESCRIPTION
**Pull request by cpaulik**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vend other packages
- [x] Build number is 0
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Fixes https://github.com/conda-forge/staged-recipes/issues/8750

### CODE REVIEW

**Comment by cpaulik** [comment]
Will solve https://github.com/conda-forge/staged-recipes/issues/8750

File: recipes/nghttp2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  skip: True  # [not unix]
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+    - pkg-config  # [unix]
```
Comment (jakirkham) [inline recipes/nghttp2/meta.yaml]: Would add an empty `host` section below. This is useful for splitting the `build` and `host` environments and ensuring there is no overlap of files between the two.

```suggestion
    - pkg-config  # [unix]
  host:
```

**Comment by jakirkham** [comment]
Please strip all of the comments from the example recipe.

**Comment by cpaulik** [comment]
I think this should work now. The recipe might be a little bit too complicated. I took the curl recipe as an example. Feedback welcome

File: recipes/nghttp2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  number: 0
+  skip: True  # [not unix]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+    - pkg-config  # [unix]
+    - zlib>=1.2.3
```
Comment (chrisburr) [inline recipes/nghttp2/meta.yaml]: These 4 should be in `host only` (only build tools like the compilers and make belong in `build`).

File: recipes/nghttp2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    - openssl>=1.0.1  # [unix]
+    - libev>=4.11
+    - c-ares>=1.7.5
+  host:
+    - zlib>=1.2.3
```
Comment (chrisburr) [inline recipes/nghttp2/meta.yaml]: The constraints need to be formatted like this (once space before the operator, none after)
```suggestion
    - zlib >=1.2.3
```

File: recipes/nghttp2/build.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig"
```
Comment (chrisburr) [inline recipes/nghttp2/build.sh]: Are these variables needed? I would expect them to be set by default

**Comment by cpaulik** [comment]
Thanks, implemented your suggestions

**Comment by jakirkham** [comment]
Thanks for working on this @cpaulik! 😄

**Pull request merged**